examples/wasm/envoy_filter_http_wasm_example.cc

```diff
@@ -1,14 +1,219 @@
 // NOLINT(namespace-envoy)
 #include <string>
+#include <vector>
 #include <unordered_map>
+#include <unordered_set>
 
+#include "nlohmann/json.hpp"
 #include "proxy_wasm_intrinsics.h"
+#include "src/libinjection.h"
+#include "src/libinjection_sqli.h"
+
+using json = nlohmann::json;
+using QueryParams = std::map<std::string, std::string>;
+
+/*
+ * Convert a config field to string
+ */
+std::string config_field_to_string(bool include, std::vector<std::string> params) {
+  std::string str = "include: ";
+  if (!include) {
+    str = "exclude: ";
+  }
+  for (auto const& s : params) {
+    str += s + ", ";
+  }
+  return str;
+}
+
+/*
+ * Keep track of config info
+ * Three main fields: param(query param), header, cookie
+ *
+ * If <field>_include is true, <field>s contains the names to be included in
+ * sql injection detection
+ * If <filed>_include is false, <field>s contains the names to be excluded
+ */
+struct Config {
+  std::string content_type { "application/x-www-form-urlencoded" };
```

Comment:
Maybe you want to have a const string somewhere for "application/x-www-form-urlencoded" as you use this here and  on lines +270 and +347

Reply:
Updated in commit [here](https://github.com/dengyijia/envoy-wasm/pull/1/commits/340940bd7a1de29740d59ef6f8e1199b4b1fa1ba). I am not sure of the recommended way to add a global variable in C++. Should I declare it within some class?